src/Search/Processors/SearchUpdateProcessor.php

```diff
@@ -83,12 +83,19 @@ protected function prepareIndexes()
                 SearchVariant::activate_state($state);
 
                 // Ensure that indexes for all new / updated objects are included
+                $indexesThatDeleted = [];
                 $objs = DataObject::get($base)->byIDs(array_keys($ids));
                 foreach ($objs as $obj) {
                     foreach ($ids[$obj->ID] as $index) {
                         if (!$indexes[$index]->variantStateExcluded($state)) {
-                            $indexes[$index]->add($obj);
-                            $dirtyIndexes[$index] = $indexes[$index];
+                            // remove any existing records from search index if ShowInSearch is false
+                            if (isset($obj->ShowInSearch) && !$obj->ShowInSearch && $obj->ID) {
```

Comment:
`isset()` means it won't fall back to `$obj->getShowInSearch()` right? Also, the check for `$obj->ID` is a bit confusing here - do we support indexing recirds without an ID (before they're saved)? I wouldn't know how we'd reference them in the index then. Seems superfluous.

Reply:
have removed the $obj->ID check

isset() does actually work for getShowInSearch() - see https://github.com/silverstripe/silverstripe-fulltextsearch/issues/257#issuecomment-601488939

Reply:
Right, we've got `ViewableData->__isset()`.